internal/fetcher/fetcher_test.go

```diff
@@ -0,0 +1,69 @@
+package fetcher
+
+import (
+	"context"
+	"github.com/databricks/databricks-sql-go/internal/config"
+	"testing"
+)
+
+// Create a mock struct for FetchableItems
+type mockFetchableItem struct {
```

Comment:
For a more realistic test we should add a wait time to mockFetchableItem and have the Fetch method sleep for that time so that we get the order of items finishing to be non-deterministic.

Reply:
Added as part of [this](https://github.com/databricks/databricks-sql-go/pull/150/commits/33fe9b15ee221575a4218b374f7753519612a7ad) commit